### PR TITLE
feat: update hcp tutorials sidebar

### PR DESCRIPTION
**Pull request by kendallstrautman**
## 🔗 Relevant links

- [Preview link](https://dev-portal-git-ksupdate-tutorials-sidebar-hashicorp.vercel.app/hcp/tutorials) 🔎
- [Asana task](https://app.asana.com/0/1199634971449915/1202932567024648) 🎟️
- [Figma Link](https://www.figma.com/file/VD7ahvXuXWJApeGnhbW4hv/Developer?node-id=12456%3A147293) 🎨 

## 🗒️ What

This PR adds a custom sidebar for the HCP tutorials landing and collection pages. 

## 📸 Design Screenshots

Screen Shot 2022-09-08 at 1 19 25 PM

## 🧪 Testing

- [Visit the preview ](https://dev-portal-git-ksupdate-tutorials-sidebar-hashicorp.vercel.app/hcp/tutorials)
- Validate that the sidebar renders correctly, matching the designs. 
- Click on a collection link,  verify that the sidebar is the same.
- Go to a tutorial, shrink down to a mobile viewport, open up the nav menu. Go up one level and verify that the custom sidebar content renders. 
- Go to other product tutorial pages to smoke-test that they are working the same. 

## 💭 Anything else?

In the future, I think we may want to refactor HCP out of the dynamic route generation that other product tutorials use, because of all the one-offs. However, leaving that large refactor out of this work to get a mvp version going.

### CODE REVIEW

File: src/views/tutorial-view/types.ts [hunk 1]
```diff
@@ -41,7 +42,9 @@ interface TutorialData
 }
 interface TutorialViewProps {
 	layoutProps: TutorialSidebarSidecarProps
-	product: LearnProductData
+	product: Omit<LearnProductData, 'slug'> & {
+		slug: LearnClientProduct['slug'] | 'hcp'
+	}
```
Comment (kendallstrautman) [inline src/views/tutorial-view/types.ts:47]: We need to do some cleanup around these types with the 'hcp' / 'cloud' mapping. This was a quick workaround as the current tutorial view didn't account for the product slug being 'hcp'.

File: src/views/collection-view/helpers/format-sidebar-sections.ts [hunk 2]
```diff
@@ -6,7 +6,7 @@ import { ListItemProps } from 'components/tutorials-sidebar/types'
 import { getCollectionSlug } from './get-slug'
 
 export interface CollectionCategorySidebarSection {
-	title?: CollectionCategoryOption
+	title?: string
```
Comment (kendallstrautman) [inline src/views/collection-view/helpers/format-sidebar-sections.ts:9]: Opening this type up as the hcp sidebar doesn't use the same collection category options and I don't think it matters much if we're being super specific on the exact string for the title.

File: src/views/tutorial-view/utils/get-canonical-collection-slug.ts [hunk 3]
```diff
@@ -14,13 +14,17 @@ import { TutorialData } from '..'
...
+		currentProductSlug === 'hcp' ? 'cloud' : currentProductSlug
 	// check if default has a beta product
 	const defaultCollectionProduct = tutorial.collectionCtx.default.slug.split(
 		'/'
-	)[0] as ProductOption | SectionOption
-	const defaultIsInBeta = getIsBetaProduct(defaultCollectionProduct)
+	)[0] as ProductOption | SectionOption | 'cloud'
+	const betaProductToCheck =
+		defaultCollectionProduct === 'cloud' ? 'hcp' : defaultCollectionProduct
+	const defaultIsInBeta = getIsBetaProduct(betaProductToCheck)
```
Comment (kendallstrautman) [inline src/views/tutorial-view/utils/get-canonical-collection-slug.ts:27]: Again this is a messy workaround for the 'hcp' / 'cloud' weirdness 🤷

File: src/components/sidebar/types.ts [hunk 4]
```diff
@@ -127,6 +128,7 @@ interface MenuItem {
 	 * icon will be shown instead.
 	 */
 	trailingIcon?: ReactElement
+	badge?: SidebarNavMenuItemBadgeProps
```
Comment (kendallstrautman) [inline src/components/sidebar/types.ts:131]: This seemed like the simplest fix for this type issue? I do see a few @TS-fixme's related to the badge stuff with the menu items, so a more permanent adjustment might be better.

File: src/views/collection-view/server.ts [hunk 5]
```diff
@@ -144,6 +153,9 @@ export async function getCollectionPagePaths(): Promise<CollectionPagePath[]> {
 	collections.forEach((collection: ClientCollection) => {
 		// assuming slug structure of :product/:filename
 		const [collectionProductSlug, filename] = collection.slug.split('/')
+
+		const productSlug =
+			collectionProductSlug === 'cloud' ? 'hcp' : collectionProductSlug
```
Comment (kendallstrautman) [inline src/views/collection-view/server.ts:158]: Bringing in a fix @BRKalow shared here to fix the collection routes.

**Review by BRKalow — APPROVED** [review]
LGTM!

**Comment by ashleemboyer** [comment]
Taking a look! 😊

File: src/views/collection-view/helpers/generate-hcp-sidebar.ts [hunk 6]
```diff
@@ -0,0 +1,72 @@
...
+	]
+
+	const categorizedNonHcpCollections = hcpContent.sidebarCategories.map(
+		(category: SidebarCategory) => ({
+			title: category.name,
+			items: category.items.map((item) => {
+				const href = getCollectionSlug(item.collectionSlug)
+				const badge = item.isBeta
+					? {
+							text: 'BETA',
```
Comment (ashleemboyer) [inline src/views/collection-view/helpers/generate-hcp-sidebar.ts:56]: Small formatting thing!

```suggestion
							text: 'Beta',
```

File: src/views/tutorial-view/index.tsx [hunk 7]
```diff
@@ -9,7 +9,11 @@ import useCurrentPath from 'hooks/use-current-path'
 import { useOptInAnalyticsTracking } from 'hooks/use-opt-in-analytics-tracking'
 import { useMobileMenu } from 'contexts'
 import InstruqtProvider from 'contexts/instruqt-lab'
-import { ProductOption, TutorialLite } from 'lib/learn-client/types'
+import {
+	ProductOption,
+	SectionOption,
+	TutorialLite,
+} from 'lib/learn-client/types'
```
Comment (ashleemboyer) [inline src/views/tutorial-view/index.tsx:16]: Looks like `SectionOption` isn't used.

```suggestion
import {	ProductOption, TutorialLite } from 'lib/learn-client/types'
```

**Review by ashleemboyer — APPROVED** [review]
Aside from the 2 small comments, this looks great! Tested both HCP and a few other Collection/Tutorial pages. ✅

**Pull request merged**